web/pandas/pdeps/0013-logical-type-system.md

```diff
@@ -0,0 +1,197 @@
+
+| Method             | NumPy-backed result | Arrow Backed result type | PDEP-13 result type            |
+|--------------------|---------------------|--------------------------|--------------------------------|
+| Series.str.len()   | np.float64          | pa.int64()               | pd.Int64Dtype()                |
+| Series.str.split() | object              | pa.list(pa.string())     | pd.ListDtype(pa.StringDtype()) |
+| Series.dt.date     | object              | pa.date32()              | pd.DateDtype()                 |
+
+The ``Series.dt.date`` example is worth an extra look - with a PDEP-13 logical type system in place we would theoretically have the ability to keep our default ``pd.DatetimeDtype()`` backed by our current NumPy-based array but leverage pyarrow for the ``Series.dt.date`` solution, rather than having to implement a DateArray ourselves.
+
+While this PDEP proposes reusing existing extension types, it also necessitates extending those types with extra metadata:
+
+```python
+class BaseType:
+
+    @property
+    def data_manager -> Literal["numpy", "pyarrow"]:
+        """
+        Who manages the data buffer - NumPy or pyarrow
+        """
+        ...
+
+    @property
+    def physical_type:
+        """
+        For logical types which may have different implementations, what is the
+        actual implementation? For pyarrow strings this may mean pa.string() versus
+        pa.large_string() versrus pa.string_view(); for NumPy this may mean object
+        or their 2.0 string implementation.
+        """
+        ...
+
+    @property
+    def na_marker -> pd.NA|np.nan|pd.NaT:
+        """
+        Sentinel used to denote missing values
+        """
+        ...
+```
+
+``na_marker`` is expected to be read-only (see next section). For advanced users that have a particular need for a storage type, they may be able to construct the data type via ``pd.StringDtype(data_manager=np)`` to assert NumPy managed storage. While the PDEP allows constructing in this fashion, operations against that data make no guarantees that they will respect the storage backend and are free to convert to whichever storage the internals of pandas considers optimal (Arrow will typically be preferred).
+
+### Missing Value Handling
+
+Missing value handling is a tricky area as developers are split between pd.NA semantics versus np.nan, and the transition path from one to the other is not always clear.
+
+Because this PDEP proposes reuse of the existing pandas extension type system, the default missing value marker will consistently be ``pd.NA``. However, to help with backwards compatibility for users that heavily rely on the equality semantics of np.nan, an option of ``pd.na_marker = "legacy"`` can be set. This would mean that the missing value indicator for logical types would be:
```

Comment:
i feel like you're not engaging with what joris and i tried to explain about "you need to do it all at once" on Wednesday's call.

Reply:
Sorry missed this before - what do you think is missing in this section?

Reply:
This section dramatically expands the scope of this PDEP, to encompass a lot of what would be PDEP-NA.  If you want to make PDEP-NA, great, go for it.  Until then, this one should either a) be tightly scoped so as to not depend on PDEP-NA, or b) be explicitly "this is on the back-burner until after PDEP-NA".  I prefer the first option, but you do you.

Reply:
Is your objection just to the default of pd.NA or to the more general approach of allowing na_marker to allow different nullability semantics while still using the same logical type? I am indifferent if we felt the need to start with the "legacy" approach of mixed sentinels, but I think the latter na_marker is a critical piece to execute this PDEP

Reply:
Do you think the na_marker="legacy" makes sense? Curious to know if you think the NA marker for logical types that have no NumPy equivalent (ex: ListDtype) should still be np.nan or pd.NA

Reply:
Bikeshedding naming aside, I'd be fine with that as long as it is the default until we do an all-at-once switchover following PDEP-NA.